activestorage/reductionist.py

```diff
@@ -65,11 +66,38 @@ def encode_slice(s):
     return [encode_slice(s) for s in selection]
 
 
+def encode_dvalue(value):
+    """Encode a data value in a JSON-compatible format."""
+    if isinstance(value, np.float32):
+        # numpy cannot encode float32, so convert it to a float64.
+        return np.float64(value)
+    return value
+
+
+def encode_missing(missing):
+    """Encode missing data in a JSON-compatible format."""
+    fill_value, missing_value, valid_min, valid_max = missing
+    # fill_value and missing_value are effectively the same when reading data.
+    missing_value = fill_value or missing_value
+    if missing_value:
+        if isinstance(missing_value, collections.abc.Sequence):
+            return {"missing_values": [encode_dvalue(v) for v in missing_value]}
```

Comment:
I was going from [this](https://docs.unidata.ucar.edu/netcdf-c/current/attribute_conventions.html) description of missing_value:

> This attribute is not treated in any special way by the library or conforming generic applications, but is often useful documentation and may be used by specific applications. The missing_value attribute can be a scalar or vector containing values indicating missing data. These values should all be outside the valid range so that generic applications will treat them as missing.

Reply:
hmm I've never seen more than one value for missing value myself, but since we now have functionality for vector/iterable we may as well keep it :+1: